modules/stereo/src/Descriptor.cpp

```diff
@@ -0,0 +1,448 @@
+						{
+							c = c << 2;
+							c = c + 1;
+						}
+						else
+						{
+							c = c << 2;
+						}
+
+					}
+					if (ii != i || jj != j)
+					{
+
+						if (image2[rrWidth + jj]  > image2[rWidth + j] - t)
+						{
+							c2 = c2 << 1;
+							c2 = c2 + 1;
+							c2 = c2 << 1;
+							c2 = c2 + 1;
+						}
+						else if (image2[rWidth + j] - t < image2[rrWidth + jj] && image2[rWidth + j] + t >= image2[rrWidth + jj])
+						{
+							c2 = c2 << 2;
+							c2 = c2 + 1;
+						}
+						else
+						{
+							c2 = c2 << 2;
+						}
+					}
+				}
+
+			}
+			for (int ii = i - n2; ii <= i + n2; ii += 4)
+			{
+				int rrWidth = ii * width;
+				for (int jj = j - n2; jj <= j + n2; jj += 4)
+				{
+					if (ii != i || jj != j)
+					{
+						if (image1[rrWidth + jj]  > image1[rWidth + j] - t)
+						{
+							c = c << 1;
+							c = c + 1;
+							c = c << 1;
+							c = c + 1;
+						}
+						else if (image1[rWidth + j] - t < image1[rrWidth + jj] && image1[rWidth + j] + t >= image1[rrWidth + jj])
+						{
+							c = c << 2;
+							c = c + 1;
+						}
+						else
+						{
+							c = c << 2;
+						}
+
+					}
+					if (ii != i || jj != j)
+					{
+
+						if (image2[rrWidth + jj]  > image2[rWidth + j] - t)
+						{
+							c2 = c2 << 1;
+							c2 = c2 + 1;
+							c2 = c2 << 1;
+							c2 = c2 + 1;
+						}
+						else if (image2[rWidth + j] - t < image2[rrWidth + jj] && image2[rWidth + j] + t >= image2[rrWidth + jj])
+						{
+							c2 = c2 << 2;
+							c2 = c2 + 1;
+						}
+						else
+						{
+							c2 = c2 << 2;
+						}
+					}
+				}
+			}
+			dist[distV + j] = c;
+			dist2[distV + j] = c2;
+		}
+	}
+}
+//the modified cs census
+void Descriptor::applyModifiedCenterSimetricCensus(uint8_t * image1, uint8_t *image2, int height, int width, int padding_top, int padding_bottom, uint32_t * &dist, uint32_t * &dist2)
+{
+	int n2 = (this->kernelSize - 1) / 2;
+
+//#pragma omp parallel for
+	for (int i = padding_top + n2; i <= height - padding_bottom - n2; i++)//sterg + 1 pt ca in c
+	{
+		int distV = (i - padding_top) * width;
+		for (int j = n2; j <= width - n2; j++)
+		{
+			uint32_t c = 0;
+			uint32_t c2 = 0;
+			for (int ii = i - n2; ii <= i + 1; ii++)
+			{
+				int rrWidth = ii * width;
+				int rrWidthC = (ii + n2) * width;
+				for (int jj = j - n2; jj <= j + n2; jj += 2)
+				{
+					if (ii != i || jj != j)
+					{
+						if (image1[rrWidth + jj] > image1[rrWidthC + (jj + n2)])
+						{
+							c = c + 1;
+						}
+						c = c * 2;
+					}
+					if (ii != i || jj != j)
+					{
+						if (image2[rrWidth + jj] > image2[rrWidthC + (jj + n2)])
+						{
+							c2 = c2 + 1;
+						}
+						c2 = c2 * 2;
+					}
+				}
+			}
+			dist[distV + j] = c;
+			dist2[distV + j] = c2;
+		}
+	}
+}
+
+//The classical center symetric census
+void Descriptor::applyCenterSimetricCensus(uint8_t * image1, uint8_t *image2, int height, int width, int padding_top, int padding_bottom, uint32_t * &dist, uint32_t * &dist2)
```

Comment:
Can you, please, consider using more "high-level" functions in implementation? For example, I'm thinking about having a single "census" function for all of the modifications (the needed modification will be specified with input parameter).